--- a/contrib/relic/src/md/blake2.h
+++ b/contrib/relic/src/md/blake2.h
@@ -61,7 +61,8 @@
     uint8_t  personal[BLAKE2S_PERSONALBYTES];  // 32
   } blake2s_param;
 
-  ALIGNME( 64 ) typedef struct __blake2s_state
+#pragma pack(pop)
+  typedef struct __blake2s_state
   {
     uint32_t h[8];
     uint32_t t[2];
@@ -69,8 +70,9 @@
     uint8_t  buf[2 * BLAKE2S_BLOCKBYTES];
     size_t   buflen;
     uint8_t  last_node;
-  } blake2s_state ;
+  } ALIGNME( 64 ) blake2s_state ;
 
+#pragma pack(push, 1)
   typedef struct __blake2b_param
   {
     uint8_t  digest_length; // 1
@@ -86,7 +88,8 @@
     uint8_t  personal[BLAKE2B_PERSONALBYTES];  // 64
   } blake2b_param;
 
-  ALIGNME( 64 ) typedef struct __blake2b_state
+#pragma pack(pop)
+  typedef struct __blake2b_state
   {
     uint64_t h[8];
     uint64_t t[2];
@@ -94,7 +97,7 @@
     uint8_t  buf[2 * BLAKE2B_BLOCKBYTES];
     size_t   buflen;
     uint8_t  last_node;
-  } blake2b_state;
+  } ALIGNME( 64 ) blake2b_state;
 
   typedef struct __blake2sp_state
   {
@@ -111,7 +114,6 @@
     uint8_t buf[4 * BLAKE2B_BLOCKBYTES];
     size_t  buflen;
   } blake2bp_state;
-#pragma pack(pop)
 
   // Streaming API
   int blake2s_init( blake2s_state *S, const uint8_t outlen );
